### PR TITLE
Add `--json` flag

### PR DESCRIPTION
**Pull request by seanmcne**
output examples  (_note: I used node directly to debug locally_)

I also added several common properties on the fast.com webpage currently to help improve data captured. 

for json human readable output
```
node .\cli.js --json-pretty -u
{
     "downloadSpeed": 29,
     "downloadUnit": "Mbps",
     "downloadedMb": 30,
     "uploadSpeed": 13,
     "uploadUnit": "Mbps",
     "uploadedMb": 40,
     "latency": 44,
     "bufferBloat": 75,
     "userLocation": "Chicago, US",
     "userIp": "1.2.3.4",
     "isDone": true
}
```

for json only output 
```
node .\cli.js --json -u        
{"downloadSpeed":32,"downloadUnit":"Mbps","downloadedMb":80,"uploadSpeed":10,"uploadUnit":"Mbps","uploadedMb":30,"latency":52,"bufferBloat":75,"userLocation":"Chicago, US","userIp":"1.2.3.4","isDone":true}
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
I don't see the point of 2 JSON flags. Make it just `--json` and pretty.

**Comment by seanmcne** [comment]
> I don't see the point of 2 JSON flags. Make it just `--json` and pretty.

I assume there are some folks that want to use this in automation and logging and wouldn't want the extra spaces and tabs, whereas some may want it human readable (it's actually matching the options that speedtest.net has for their CLI).

**Comment by sindresorhus** [comment]
> I assume there are some folks that want to use this in automation and logging and wouldn't want the extra spaces and tabs, whereas some may want it human-readable (it's actually matching the options that speedtest.net has for their CLI).

Logging is a good point. Let's only do the one-line version then. I don't really see the point of human-readable JSON output anyway, as we already have a nice output for humans. If someone needs a nice JSON output they can pipe the output through some JSON formatter.

**Comment by sindresorhus** [comment]
> My previous experience with a similar command line tool was for speedtest-cli - here is the command help output for that cli tool which is what I patterned json and json-pretty from:

I prefer to keep my CLIs minimal.

**Comment by seanmcne** [comment]
> > I assume there are some folks that want to use this in automation and logging and wouldn't want the extra spaces and tabs, whereas some may want it human-readable (it's actually matching the options that speedtest.net has for their CLI).
> 
> Logging is a good point. Let's only do the one-line version then. I don't really see the point of human-readable JSON output anyway, as we already have a nice output for humans. If someone needs a nice JSON output they can pipe the output through some JSON formatter.

Sounds good - I just adjusted cli.js.  Let me know what you thank - thx!

File: cli.js [hunk 1]
```diff
@@ -72,6 +79,10 @@ const speed = () => speedText() + lineBreak(2);
 function exit() {
 	if (process.stdout.isTTY) {
 		logUpdate(`${lineBreak(2)}${spacing(4)}${speed()}`);
+	}
+
+	if (cli.flags.json) {
+		logUpdate(JSON.stringify(data, null, null));
```
Comment (sindresorhus) [inline cli.js:85]:
```suggestion
		logUpdate(JSON.stringify(data));
```

File: cli.js [hunk 2]
```diff
@@ -47,8 +54,8 @@ dns.lookup('fast.com', error => {
 let data = {};
 const spinner = ora();
 
-const lineBreak = amount => (cli.flags.singleLine ? '' : '\n'.repeat(amount));
-const spacing = amount => (cli.flags.singleLine ? '' : ' '.repeat(amount));
+const lineBreak = amount => (cli.flags.singleLine || cli.flags.json || cli.flags.jsonPretty ? '' : '\n'.repeat(amount));
+const spacing = amount => (cli.flags.singleLine || cli.flags.json || cli.flags.jsonPretty ? '' : ' '.repeat(amount));
```
Comment (sindresorhus) [inline cli.js:58]: Outdated code

File: cli.js [hunk 3]
```diff
@@ -15,11 +15,15 @@ const cli = meow(`
 	Options
 	  --upload, -u   Measure upload speed in addition to download speed
 	  --single-line  Reduce spacing and output to a single line
+	  --json         Process output in json format also forces single line
```
Comment (sindresorhus) [inline cli.js:18]:
```suggestion
	  --json         JSON output
```

You also need to update the readme.

File: cli.js [hunk 4]
```diff
@@ -15,11 +15,15 @@ const cli = meow(`
...
 	  --upload, -u   Measure upload speed in addition to download speed
 	  --single-line  Reduce spacing and output to a single line
+	  --json         Process output in json format also forces single line 
 
 	Examples
 	  $ fast --upload > file && cat file
 	  17 Mbps
 	  4.4 Mbps
+
+	  $ fast --upload --json-pretty
```
Comment (sindresorhus) [inline cli.js:25]: Outdated

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
@seanmcne I would recommend looking over your own diff after doing changes. I think you could have caught some of the things I commented yourself. 👍🏻

**Comment by sindresorhus** [comment]
Closing in favor of #58

**Pull request closed without merging**